pkg/datapath/loader/cache.go

```diff
@@ -183,10 +183,8 @@ func (o *objectCache) fetchOrCompile(ctx context.Context, nodeCfg *datapath.Loca
 	obj := o.serialize(hash)
 	defer obj.Unlock()
 
-	if cached, err := os.Open(obj.path); err == nil {
-		return cached, false, nil
-	} else if !errors.Is(err, os.ErrNotExist) {
-		return nil, false, err
+	if obj.spec != nil {
+		return obj.spec.Copy(), false, nil
```

Comment:
why does this return `compiled == false`?

The method comment says `... and whether the cache was hit`, this seems like a cache hit?

Reply:
Good point, the doc comment is incorrect because it inverts the condition. Fixed.